docs/openapi/openapi.yaml

```diff
@@ -0,0 +1,118 @@
+openapi: "3.0.2"
+info:
+  title: LPA ID Service
+  version: "1.0"
+  license:
+    name: MIT
+    url: https://opensource.org/licenses/MIT
+x-amazon-apigateway-policy:
+  Version: "2012-10-17"
+  Statement:
+    - Effect: Allow
+      Principal:
+        AWS: ${allowed_roles}
+      Action: "execute-api:Invoke"
+      Resource:
+        - ${lambda_invoke_arn}
+security:
+  - {}
+paths:
+  /cases:
+    post:
+      operationId: createCase
+      summary: Generate a case ID
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - type
+                - source
+                - donor
+              properties:
+                type:
+                  type: string
+                  enum:
+                    - "hw"
+                    - "pfa"
+                source:
+                  type: string
+                  enum:
+                    - "APPLICANT"
+                    - "PHONE"
+                donor:
+                  type: object
+                  required:
+                    - name
+                    - dob
+                    - postcode
+                  properties:
+                    name:
+                      type: string
+                      example: Jacalyn Petralba
```

Comment:
should name be split into firstname, middlename(s), surname?

Reply:
I wasn't sure that we'd care. If someone rings up over the phone and we're just sending them an initial letter, then I guess we just want a contact name rather than anything formal. I figured it'd be better to keep it simple like this until we know that we need structured data.

That said, all of this structure is a bit of a guess until we start designing things to integrate with it 😅